### PR TITLE
Bump sra-tools to 2.10.3

### PR DESCRIPTION
**Pull request by mvdbeek**
**Replace this sentence with a detailed description of your pull request**

----

> Bioconda requires reviews prior to merging pull-requests (PRs). To facilitate this, once your PR is passing tests and ready to be merged, please add the `please review & merge` label so other members of the bioconda community can have a look at your PR and either make suggestions or merge it.

* [ ] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/contributor/guidelines.html).
* [ ] This PR adds a new recipe.
* [ ] AFAIK, this recipe **is directly relevant to the biological sciences**
      (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

> Everyone has access to the following BiocondaBot commands, which can be given in a comment:
>
>  * `@BiocondaBot please update` will cause the BiocondaBot to merge the master branch into a PR
>  * `@BiocondaBot please add label` will add the `please review & merge` label.
>  * `@BiocondaBot please fetch artifacts` will post links to packages and docker containers built by the CI system. You can use this to test packages locally before merging.
>
> For members of the Bioconda project, the following command is also available:
>
>  * `@BiocondaBot please merge` will cause packages/containers to be uploaded and a PR merged. Someone must approve a PR first! This has the benefit of not wasting CI build time required by manually merging PRs.
>
> If you have questions, please post them in gitter or ping `@bioconda/core` in a comment (if you are not able to directly ping `@bioconda/core` then the bot will repost your comment and enable pinging).

### CODE REVIEW

File: recipes/sra-tools/meta.yaml [hunk 1]
```diff
@@ -27,6 +27,7 @@ requirements:
   host:
     - ncbi-ngs-sdk
     - libxml2
+    - libutf8proc
```
Comment (mvdbeek) [inline recipes/sra-tools/meta.yaml]: That's not the problem, they bundle it. I think something in the build process is weird.

Comment (mvdbeek) [reply recipes/sra-tools/meta.yaml]:
```
09:47:55 BIOCONDA INFO (OUT) make[6]: Entering directory `$SRC_DIR/sra-tools/tools/driver-tool/utf8proc'
09:47:55 BIOCONDA INFO (OUT) gcc -c  -std=gnu11  -DNDEBUG -m64   -DLINUX -DUNIX -D_GNU_SOURCE -D_REENTRANT -D_FILE_OFFSET_BITS=64 -DPKGNAME=linux64 -D_ARCH_BITS=__SIZEOF_POINTER__*__CHAR_BIT__ -DLIBPREFIX=lib -DSHLIBEXT=so  -I$SRC_DIR/sra-tools/interfaces/override -I$SRC_DIR/sra-tools/tools/driver-tool/json/linux -I$SRC_DIR/sra-tools/tools/driver-tool/json/unix -I$SRC_DIR/sra-tools/tools/driver-tool/json -I$SRC_DIR/ncbi-vdb/interfaces/ -I$SRC_DIR/ncbi-vdb/interfaces/cc/gcc/x86_64 -I$SRC_DIR/ncbi-vdb/interfaces/cc/gcc -I$SRC_DIR/ncbi-vdb/interfaces/os/linux -I$SRC_DIR/ncbi-vdb/interfaces/os/unix -I$SRC_DIR/ncbi-vdb/interfaces/ext -I$PREFIX/include -I. -MD -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem $PREFIX/include -std=gnu11 -DNDEBUG -m64 -DLINUX -DUNIX -D_GNU_SOURCE -D_REENTRANT -D_FILE_OFFSET_BITS=64 -DPKGNAME=linux64 -D_ARCH_BITS=__SIZEOF_POINTER__*__CHAR_BIT__ -DLIBPREFIX=lib -DSHLIBEXT=so -I$SRC_DIR/sra-tools/interfaces/override -I$SRC_DIR/sra-tools/tools/driver-tool/json/linux -I$SRC_DIR/sra-tools/tools/driver-tool/json/unix -I$SRC_DIR/sra-tools/tools/driver-tool/json -I$SRC_DIR/ncbi-vdb/interfaces/ -I$SRC_DIR/ncbi-vdb/interfaces/cc/gcc/x86_64 -I$SRC_DIR/ncbi-vdb/interfaces/cc/gcc -I$SRC_DIR/ncbi-vdb/interfaces/os/linux -I$SRC_DIR/ncbi-vdb/interfaces/os/unix -I$SRC_DIR/ncbi-vdb/interfaces/ext -I$PREFIX/include -I. -fPIC -std=c99 -Wall -pedantic -DUTF8PROC_EXPORTS -c -o utf8proc.o utf8proc.c
09:47:56 BIOCONDA INFO (OUT) rm -f libutf8proc.a
09:47:56 BIOCONDA INFO (OUT) ar rc rs libutf8proc.a utf8proc.o
09:47:56 BIOCONDA INFO (OUT) ar: libutf8proc.a: No such file or directory
09:47:56 BIOCONDA INFO (OUT) make[6]: *** [libutf8proc.a] Error 1
09:47:56 BIOCONDA INFO (OUT) make[6]: Leaving directory `$SRC_DIR/sra-tools/tools/driver-tool/utf8proc'
09:47:56 BIOCONDA INFO (OUT) make[5]: *** [/opt/conda/conda-bld/sra-tools_1582710149751/work/sra-tools/tools/driver-tool/utf8proc/utf8proc.o] Error 2
09:47:56 BIOCONDA INFO (OUT) make[5]: Leaving directory `$SRC_DIR/ncbi-outdir/sra-tools/linux/gcc/x86_64/rel/obj/tools/driver-tool/json'
09:47:56 BIOCONDA INFO (OUT) make[4]: *** [std] Error 2
09:47:56 BIOCONDA INFO (OUT) make[4]: Leaving directory `$SRC_DIR/sra-tools/tools/driver-tool/json'
09:47:56 BIOCONDA INFO (OUT) make[3]: *** [/opt/conda/conda-bld/sra-tools_1582710149751/work/ncbi-outdir/sra-tools/linux/gcc/x86_64/rel/ilib/libjson.a] Error 2
09:47:56 BIOCONDA INFO (OUT) make[3]: Leaving directory `$SRC_DIR/ncbi-outdir/sra-tools/linux/gcc/x86_64/rel/obj/tools/driver-tool'
09:47:56 BIOCONDA INFO (OUT) make[2]: *** [std] Error 2
09:47:56 BIOCONDA INFO (OUT) make[2]: Leaving directory `$SRC_DIR/sra-tools/tools/driver-tool'
09:47:56 BIOCONDA INFO (OUT) make[1]: *** [driver-tool] Error 2
09:47:56 BIOCONDA INFO (OUT) make[1]: Leaving directory `$SRC_DIR/sra-tools/tools'
09:47:56 BIOCONDA INFO (OUT) make: *** [tools] Error 2
```
with that commit. Maybe there is a flag to not use the bundled version ?

**Comment by raivivek** [comment]
@mvdbeek Version 2.10.4 is out so probably best to just work on that. What do you think?
https://github.com/bioconda/bioconda-recipes/pull/19764

**Comment by bgruening** [comment]
@raivivek probably. But the other version has, unfortunately, the same problem :(

**Comment by raivivek** [comment]
@bgruening I've found the bug I think. Will fix it soon.

**Comment by bgruening** [comment]
But you got it to build on Linux! Awesome! Maybe we skip OSX for the time being?

**Comment by raivivek** [comment]
@bgruening Yeah, I have other issues open to track the macOS problem, so skipping osx for now is not a bad idea. Thanks!

**Comment by raivivek** [comment]
@bioconda-bot please fetch artifacts

**Comment by raivivek** [comment]
@bioconda-bot please merge

**Comment by dukechem** [comment]
brew can build ok from source for me on macos 10.14  ... see transcript at [https://github.com/ncbi/sra-tools/issues/299#issuecomment-700849804](https://github.com/ncbi/sra-tools/issues/299#issuecomment-700849804)

**Pull request merged**